### PR TITLE
Add SASL Parameter

### PR DESCRIPTION
**Pull request by MikkelVestergaard**
This change will make it possible to switch authentication mechanism
from PLAIN to EXTERNAL as described here:
https://www.rabbitmq.com/access-control.html#mechanisms
These are the two standard authentication mechanisms.

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClientConfig.java [hunk 1]
```diff
@@ -136,6 +136,12 @@
     @ConfigItem(defaultValue = "true")
     public boolean topologyRecovery;
 
+    /**
+     * SASL authentication mechanisms
+     */
+    @ConfigItem
+    public String sasl;
```
[the remark this answers is not in the document]
Comment (MikkelVestergaard) [reply runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQClientConfig.java]: I added the default value as an enum

File: runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQHelper.java [hunk 2]
```diff
@@ -40,6 +40,14 @@ public static Connection newConnection(RabbitMQClientParams params, String name)
     private static ConnectionFactory newConnectionFactory(RabbitMQClientParams params) {
         ConnectionFactory cf = new ConnectionFactory();
         cf.setSharedExecutor(params.getExecutorService());
+
+        final String sasl = params.getConfig().sasl;
+        if ("EXTERNAL".equals(sasl)) {
+            cf.setSaslConfig(DefaultSaslConfig.EXTERNAL);
+        } else if ("PLAIN".equals(sasl)) {
```
[the remark this answers is not in the document]
Comment (MikkelVestergaard) [reply runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQHelper.java]: I changed it to an enum and moved the mapping to the new enum.

**Comment by MikkelVestergaard** [comment]
hi @bpasson
Sorry for the very slow response.
I made the suggest change to an enum and added the parameter to the testcase.

**Comment by bpasson** [comment]
@MikkelVestergaard I will have a look.

**Comment by bpasson** [comment]
I am gonna merge it, and change the default value from `PLAIN` to `plain` as lowercase enums work too and just look a bit better in the documentation.

**Comment by bpasson** [comment]
@all-contributors please add @MikkelVestergaard  for code

**Pull request merged**